docs/service-activity.md

```diff
@@ -0,0 +1,21 @@
+# Service Activity Framework
+
+The service activity framework is a simple framework allowing pluggable go-routine powered extension points. All of the features of `stencil-golang` are implemented using this framework (`grpc`, `http`, etc).
+
+## How it Works
+
+The framework is implemented by using the [`async.Runner`](https://github.com/getoutreach/gobox/blob/c8a5b56788ef7a8ad27373c63febf6f993e1a1b0/pkg/async/async.go#L48) interface for every activity. ([`async.Closer`](https://github.com/getoutreach/gobox/blob/c8a5b56788ef7a8ad27373c63febf6f993e1a1b0/pkg/async/async.go#L53) is also implemented for activities that need to close resources.). Each activity is started in [`async.RunGroup`](https://github.com/getoutreach/gobox/blob/c8a5b56788ef7a8ad27373c63febf6f993e1a1b0/pkg/async/async.go#L136) and waits for the following conditions:
+
+ * An activity errors
+ * Context is canceled
+ * SIGINT is received
+
+When any of those conditions are reached, all activities' are stopped and their `Close()` function (if applicable) is called. It is up to the activities to gracefully shutdown. However, as a special case, if `SIGINT` is received a second time, the program is forcefully terminated. This allows users/service runners to shutdown when needed.
+
+## Adding a custom service activity
+
+<!-- TODO(jaredallard): It'd be nice to have a full tutorial here, but for now this is good -->
```

Comment:
I'm actually going to punt this until _after_ smartstore stuff is moved out. Otherwise the docs will be out of date because the interfaces are going to change

Reply:
nit but if you don't want to forget you should create a ticket then